x-pack/plugins/canvas/canvas_plugin_src/uis/arguments/filter_group.js

```diff
@@ -21,7 +21,13 @@ const FilterGroupInput = ({ onValueChange, argValue, argId, filterGroups }) => {
   const [inputValue, setInputValue] = useState('');
   const [addMode, setAddMode] = useState(false);
 
-  const choices = [{ text: 'No group', value: '' }].concat(filterGroups.map(f => ({ text: f })));
+  // make sure the argValue is always included in the filter group list
+  const argValueChoice = !filterGroups.includes(argValue) ? [{ text: argValue }] : [];
```

Comment:
Almost there! This does fix the missing group in the select field, but it also adds an empty option when you haven't set a group yet. This should fix it
```suggestion
  const argValueChoice = argValue && !filterGroups.includes(argValue) ? [{ text: argValue }] : [];
```